## The task

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file Data/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other

In [43]:
import statistics as st
from scipy import stats
import numpy as np

In [44]:
# My data

new_machine = [42.1, 41, 41.3, 41.8, 42.4, 42.8, 43.2, 42.3, 41.8, 42.7]
old_machine = [42.7, 43.6, 43.8, 43.3, 42.5, 43.5, 43.1, 41.7, 44, 44.1]

In [45]:
# Calculating the mean for both variable.

new_machine_mean = np.mean(new_machine)
new_machine_mean

42.14

In [46]:
old_machine_mean = round(np.mean(old_machine), 2)
old_machine_mean

43.23

In [47]:
# Calculating the standard deviations for both variable.

std_new = round(np.std(new_machine), 3)
std_new

0.648

In [48]:
std_old = round(np.std(old_machine), 3)
std_old

0.711

### The hypothesis

In [49]:
# H0: The new machine works as fast as the current machine
# Ha: New machine packs faster than the current machine.

In [50]:
# It is a two sample one tail test.

### The test statistic and confidence level.

In [51]:
# I will assume there is enough evidence for the t test, our sample size is < 30. I will use t test.

# ɑ = 0.05

# n_new = 10
# n_old = 10

In [52]:
# Calculating the t value.

t = round((old_machine_mean - new_machine_mean) / np.sqrt(((std_old**2)/10) + ((std_new**2)/10)), 4)
t

3.5831

In [53]:
# Calculating the critical value.

tc = round(+-stats.t.ppf(0.95, df=9), 4)
tc

-1.8331

### Final decision

Our test statistic is 3.5831, which is larger than the critical value 1.8331. Therefore, I will reject the H0 and accept the Ha - the new machine packs faster than the old machine.